libafl/src/events/mod.rs

```diff
@@ -452,13 +448,26 @@ pub trait EventFirer: UsesState {
 }
 
 /// [`ProgressReporter`] report progress to the broker.
-pub trait ProgressReporter: EventFirer
-where
-    Self::State: HasMetadata + HasExecutions + HasLastReportTime,
-{
```

Comment:
This is the kind of change I was talking about, @domenukk. We can just straight up avoid all these generics here and shift them to impls.